src/components/Tab.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import * as Conseil from '../Conseil';
+import { ConseilFilter } from '../Conseil';
+import { DataView } from '../types';
+import { ExpandableGrid } from './ExpandableGrid';
+
+interface TabProps {
+    dataView: DataView;
+    hidden: boolean;
+    filters: ConseilFilter;
+    network: string;
+}
+
+interface TabState {
+    data: Object[];
+}
+
+export class Tab extends React.Component<TabProps, TabState> {
+
+    constructor(props: TabProps) {
+        super(props);
+        this.state = {data: []};
+        this.refreshData = this.refreshData.bind(this);
+        this.refreshData(props);
+    }
+
+    refreshData(nextProps: TabProps) {
+        if (!nextProps.hidden) {
+            let funcToCall = Conseil.getBlocks;
+            if (this.props.dataView === DataView.Operations) {funcToCall = Conseil.getOperations; }
+            if (this.props.dataView === DataView.Accounts) {funcToCall = Conseil.getAccounts; }
+            funcToCall.apply(null, [this.props.network, nextProps.filters]).
+            then((val) => this.setState({data: JSON.parse(decodeURI(val))}));
+        }
+    }
+
+    componentWillReceiveProps(nextProps: TabProps) {
```

Comment:
I think you could just have `refreshData` fire in the render function. I think the prop change should trigger a rerender, then you could delete this `componentWillReceiveProps`...

Reply:
Invoking refreshData() from render() caused it to be called infinitely many times as the former resets the state each time it is called. Is there a better way to prevent this?

Reply:
Ok an alternative is to move these calls up to the parent and invoke them `onClick` of each corresponding view button (i.e. make the call in `DataPanel.tsx`), then pass the data down as a prop to each of the tabs. Then can pass each particular async call to each button rather than having a general `funcToCall` that changes depending on the data view. Does that make sense?

Have this button click invoke the async function:
```
<button onClick={() => props.switchTab(DataView.Blocks)}>Blocks</button>
```

Reply:
I'm not fully convinced as it seems to me the data displayed by the tab is local to its state. The higher level container would otherwise always have to know how to query the data for all its children. This way, the higher level container can have all kinds of children without knowing much about them. They would simply be bound by a common minimal vocabulary, i.e. the filters. Let me know what you think.

Reply:
That's fair, I'll need to think about this. There must be a way to do this without `componentWillReceiveProps` https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops